analysers/Analyser_Osmosis.py

```diff
@@ -105,6 +105,93 @@ class Analyser_Osmosis(Analyser):
 ;
 
 ANALYZE {0}.highway_ends;
+"""
+
+    sql_create_multipolygons = """
+DO $$
+DECLARE
+    mp RECORD;
+BEGIN
+    DROP TABLE IF EXISTS {0}.multipolygons;
+    CREATE UNLOGGED TABLE {0}.multipolygons (
+        LIKE relations INCLUDING ALL,
+        poly geometry(Geometry, 4326) NOT NULL,
+        poly_proj geometry(Geometry, {1}) NOT NULL,
+        is_valid boolean NOT NULL
+    );
+
+    FOR mp in (
+        SELECT
+            relations.*,
+            ST_LineMerge(ST_Collect(ways.linestring)) AS linestrings
+        FROM
+            relations
+            JOIN relation_members ON
+                relation_members.relation_id = relations.id AND
+                relation_members.member_type = 'W' AND
+                relation_members.member_role IN ('outer', 'inner')
+            JOIN ways ON
+                ways.id = relation_members.member_id
```

Comment:
If I understand https://github.com/osm-fr/osmose-backend/issues/1818#issuecomment-1522291312 correctly, the multipolygon will be incomplete at boundaries. This could cause false positives for area-based queries (especially with multiple outers) and similar kind of queries (i.e. if we use is_valid in analyser_osmosis_relation_multipolygon to detect bad ones).
The current implementation will not abort/fail if a member way is missing it seems, it just pretends that a non-downloaded way doesn't exist.

Is it possible (and not too slow) to use a `LEFT JOIN` instead of a regular `JOIN` with ways?
Then the `SELECT` could contain `COUNT(relation_id) = COUNT(ways.id) AS is_complete` (which could be added as a row to the final table)

That way, analysers where borders may cause issues can simply query on `multipolygon.is_complete`. 
Otherwise, each analyser where this is relevant would need to go through the following (dummy) code to filter such relations out:
```sql
SELECT
  mp.*
FROM
  multipolygons AS mp
  JOIN relation_members ON
    relation_members.relation_id = mp.id AND
    relation_members.member_type = 'W'
  LEFT JOIN ways ON
    ways.id = member_id
HAVING
    COUNT(mp.id) = COUNT(ways.id)
```

Reply:
> Is it possible (and not too slow) to use a LEFT JOIN instead of a regular JOIN with ways?

Right added in the todo list.

> Then the SELECT could contain COUNT(member_id) = COUNT(ways.id) AS is_complete (which could be added as a row to the final table)

I prefer to not include incomplete MP.